init.c

```diff
@@ -955,6 +956,13 @@ int luaopen_libcutorch(lua_State *L)
   luaL_setfuncs(L, cutorch_stuff__, 0);
 
   THCState* state = (THCState*)malloc(sizeof(THCState));
+  memset(state, 0, sizeof(THCState));
+
+  char* thc_caching_allocator = getenv("THC_CACHING_ALLOCATOR");
```

Comment:
This is specific to the Lua API. Other interfaces to THC may want to always use the caching allocator (or swap the defaults so that it's enabled by default and requires the env variable to disable it).